proto/groups/v1alpha/groups.proto

```diff
@@ -0,0 +1,204 @@
+syntax = "proto3";
+
+package com.mintter.groups.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "mintter/backend/genproto/groups/v1alpha;groups";
+
+// Groups service exposes the functionality for managing group entities.
+// This API uses fully-qualified EID as group IDs in all the requests.
+service Groups {
+  // Creates a new group.
+  rpc CreateGroup(CreateGroupRequest) returns (Group);
+
+  // Gets a group by ID.
+  rpc GetGroup(GetGroupRequest) returns (Group);
+
+  // Updates a group.
+  rpc UpdateGroup(UpdateGroupRequest) returns (Group);
+
+  // Lists members of a group.
+  rpc ListMembers(ListMembersRequest) returns (ListMembersResponse);
+
+  // Lists content of a group.
+  rpc ListContent(ListContentRequest) returns (ListContentResponse);
+
+  // Lists groups.
+  rpc ListGroups(ListGroupsRequest) returns (ListGroupsResponse);
+}
+
+// Request to create a group.
+message CreateGroupRequest {
+  // Required. Title of the new group.
+  string title = 1;
+
+  // Optional. Description of the new group.
+  string description = 2;
+
+  // Optional. List of initial members for the new group.
+  // Members can also be managed with separate requests after group is already created.
+  map<string, Role> members = 3;
+}
+
+// Request to get a group.
+message GetGroupRequest {
+  // Required. ID of the group to get.
+  // Must be a fully-qualified EID.
+  string id = 1;
+
+  // Optional. Version of the group to get information about.
+  // If empty, latest_version is will be used.
+  string version = 2;
+}
+
+// Request to update a group.
+// Some fields are required, even if unchanged,
+// to distinguish between setting the value to an empty string, from an unchanged value.
+// Read the comments!
+message UpdateGroupRequest {
+  // ID of the group to update.
+  string id = 1;
+
+  // Required. Title of the Group.
+  // Must always be provided, even if unchanged.
```

Comment:
Why can't we just fail the request if the title is blank? A blank title makes no sense (unlike a blank description that it could be understood as laziness from the owner)

Reply:
That's right. We can leave the title as optional, because setting title to empty doesn't make sense. If we say that it's the same for description, we could make it optional too. Although clearing a description might be a legitimate use case.